src/main/java/org/kohsuke/github/GHCreateRepositoryBuilder.java

```diff
@@ -131,6 +132,20 @@ public GHCreateRepositoryBuilder fromTemplateRepository(String templateOwner, St
         return this;
     }
 
+    /**
+     * Create repository from template repository.
+     *
+     * @param templateRepository
+     *            the template repository as a GHRepository
+     * @return a builder to continue with building
+     * @see <a href="https://developer.github.com/v3/previews/">GitHub API Previews</a>
+     */
+    @Preview(BAPTISTE)
+    public GHCreateRepositoryBuilder fromTemplateRepository(GHRepository templateRepository) {
+        Objects.requireNonNull(templateRepository, "templateRepository cannot be null");
```

Comment:
You could also add a check that the repository in the parameter is a template repository.